### ISSUE TITLE
TypeError when using "c-cpp-flylint.includePaths"

### ISSUE THREAD

**Issue by twdkeule** [jbenden/vscode-c-cpp-flylint#10]
I specified some paths in "c-cpp-flylint.includePaths" in the workspace settings, but this triggered a TypeError in the plugin: 
`'Cannot read property 'length' of undefined' while validating: /.../SOME_CPP_FILE.cpp.`

```
shell.ts:423 Cannot read property 'document' of undefined: TypeError: Cannot read property 'document' of undefined
    at vscode.window.onDidChangeTextEditorSelection (/home/$USER/.vscode/extensions/ms-vscode.cpptools-0.12.3/out/src/LanguageServer/C_Cpp.js:121:97)
    at e.invoke (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:55038)
    at e.fire (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:56829)
    at t.$acceptSelectionsChanged (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:394258)
    at t.e.handle (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:308751)
    at s (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:188848)
    at h (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:23:189503)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
e.onUnexpectedError @ shell.ts:423
(anonymous) @ shell.ts:383
e.onUnexpectedError @ errors.ts:119
o @ errors.ts:139
t.onUnexpectedExtHostError @ mainThreadErrors.ts:13
e.handle @ abstractThreadService.ts:34
s @ ipcRemoteCom.ts:275
f @ ipcRemoteCom.ts:231
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
messageService.ts:125 vscode-c-cpp-flylint: 'Cannot read property 'length' of undefined' while validating: /.../file.cpp. Please analyze the 'C/C++ FlyLint' Output console. Stacktrace: TypeError: Cannot read property 'length' of undefined
    at makeDiagnostic (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/server.js:171:36)
    at lintersCopy.forEach.linter (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/server.js:104:38)
    at Array.forEach (native)
    at validateTextDocument (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/server.js:99:17)
    at documents.onDidSave (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/server.js:59:9)
    at CallbackList.invoke (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/node_modules/vscode-languageserver/node_modules/vscode-jsonrpc/lib/events.js:114:39)
    at Emitter.fire (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/node_modules/vscode-languageserver/node_modules/vscode-jsonrpc/lib/events.js:178:36)
    at connection.onDidSaveTextDocument (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/node_modules/vscode-languageserver/lib/main.js:191:33)
    at handleNotification (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/node_modules/vscode-languageserver/node_modules/vscode-jsonrpc/lib/main.js:474:43)
    at processMessageQueue (/home/$USER/.vscode/extensions/jbenden.c-cpp-flylint-0.2.5/server/node_modules/vscode-languageserver/node_modules/vscode-jsonrpc/lib/main.js:246:17)
e.doShow @ messageService.ts:125
e.show @ messageService.ts:104
(anonymous) @ mainThreadMessageService.ts:64
n.Class.derive._oncancel @ winjs.base.raw.js:1641
t.showMessage @ mainThreadMessageService.ts:34
t.$showMessage @ mainThreadMessageService.ts:28
e.handle @ abstractThreadService.ts:34
s @ ipcRemoteCom.ts:275
f @ ipcRemoteCom.ts:231
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
```

**Comment by jbenden** [comment]
Hello,

Please ensure that your settings match the structure below, as shown in one of my own configurations:

```
// Place your settings in this file to overwrite the default settings
{
    "c-cpp-flylint.debug": true,
    "c-cpp-flylint.language": "c",
    "c-cpp-flylint.standard": [
        "c99"
    ],
    "c-cpp-flylint.includePaths": [
        ".",
        "./zlib",
        "./popt"
    ],
    "c-cpp-flylint.defines": [
        "NDEBUG",
        "HAVE_CONFIG_H"
    ],
    "c-cpp-flylint.clang.executable": "clang-3.9",
    "c-cpp-flylint.flexelint.enable": false,
    "c-cpp-flylint.run": "onType"
}
```

If this does not resolve your problem observed, then please post your configuration settings.

**Comment by twdkeule** [comment]
Yes the `includePaths` is a regular array.

**Comment by jbenden** [comment]
Please post your configuration settings, operating system, the output of `uname -a` (if a POSIX system); so I may further assist.

**Comment by twdkeule** [comment]
Ubuntu 17.04: 4.10.0-32-generic #36-Ubuntu SMP Tue Aug 8 12:10:06 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
"c-cpp-flylint.includePaths": [
		"/usr/include/c++/6",
		"/usr/include/x86_64-linux-gnu/c++/6",
		"/usr/local/include",
		"/usr/include",
		"/usr/include/linux",
		"/usr/include/x86_64-linux-gnu",
		"/home/.../build_scripts/pc_build/usr/include",
		"/home/.../build_scripts/pc_build/usr/include/stlplus3",
		"/home/.../build_scripts/pc_build/usr/include/unicos",
		"/home/.../build_scripts/pc_build/usr/include/rmtp",
		"/home/.../unicos_common",
		"/home/.../unicos_cu",
		"${workspaceRoot}"
	]
```

**Comment by jbenden** [comment]
After some research, I have found that the document being scanned has somehow generated a line number which is not correctly found (a bad index), and is thus undefined - which the next code attempts to retrieve the line's string length - resulting in throwing this TypeError.

Is it possible for you to share the source code document that produces this TypeError?

If not, I can "fix" the TypeError by checking for the line resulting in an `undefined` value; which will solve the thrown error, but it does not explain (and thus I cannot solve) how it came to be an invalid line number within the document in the first place.

Thank you for your time and patience! I look forward to your response.

**Comment by jbenden** [comment]
I have published a new release which should solve the problems experienced.

Would you mind updating to the latest release (v0.2.6), testing your project's workspace, and reporting back your test results?

Thank you for your time and patience!

**Comment by twdkeule** [comment]
Everything seems to be working now!
